### DISCUSSION TITLE
A 'BarrierFunctionMaterial' is not a registered object.

### DISCUSSION THREAD

**Discussion by Leiming-123** [idaholab/moose#24955] [Q&A General]
Hi everyone,
I try to run a input file by using moose. But it shows" 
```
*** ERROR ***
A 'BarrierFunctionMaterial' is not a registered object.
If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable."
```

Could anyone please tell me how to solve this problem?
I download the code from the "**https://nanohub.org/resources/tincopperem**". And I select phase-field modules and try to run the "coarse_eightgrainEM_0_BC_commentless.i"

**Comment by GiudGiud** [comment]
Hello

What `ldd <your executable>` return ? (only works on linux, let me know if you are on macos)

Guillaume
